File: system-tests/debug-logging-default-test/src/test/java/com/teamscale/client/DebugLoggingDefaultTest.java
package com.teamscale.client;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.util.Optional;
import java.util.stream.Stream;

import static org.assertj.core.api.Assertions.assertThat;

public class DebugLoggingDefaultTest {

	@Test
	public void systemTest() throws Exception {
		Path logDirectory = searchForAgentTempDirectory().resolve("logs");
		assertThat(logDirectory).exists();
		assertThat(Files.readAllLines(logDirectory.resolve("teamscale-jacoco-agent.log"))).isNotEmpty();
	}

	private Path searchForAgentTempDirectory() throws IOException {
		Path tempDirectory = Paths.get(System.getProperty("java.io.tmpdir"));
		try (Stream<Path> stream = Files.list(tempDirectory)) {
			Optional<Path> maybeTempDirectory = stream.
					filter(path -> path.getFileName().toString().contains("teamscale-java-profiler"))
					.sorted().findFirst();
			return maybeTempDirectory.orElseThrow(() -> new AssertionError("Could not locate agent temp directory"));
		}
	}

}

File: system-tests/debug-logging-default-test/src/test/kotlin/com/teamscale/client/DebugLoggingDefaultTest.kt
package com.teamscale.client

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test
import java.nio.file.Paths
import kotlin.io.path.exists
import kotlin.io.path.readLines

class DebugLoggingDefaultTest {

	@Test
	fun systemTest() {
		val logDirectory = searchForAgentTempDirectory().resolve("logs")
		assertThat(logDirectory.exists()).isTrue()
		assertThat(logDirectory.resolve("teamscale-jacoco-agent.log").readLines()).isNotEmpty()
	}

	private fun searchForAgentTempDirectory() =
		Paths.get(System.getProperty("java.io.tmpdir"))
			.toFile().walk().maxDepth(1)
			.firstOrNull { it.name.contains("teamscale-java-profiler") }?.toPath()
			?: throw AssertionError("Could not locate agent temp directory")
}
